### PR TITLE
Improve toString, add exception message, stacktrace processing and improved readme

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

**Comment by whiskeysierra** [comment]
@lukasniemeier-zalando Can you check if this solves your issues #10 and #11?
@codechimp Would you say the modified readme is now clear enough regarding the dependency to the `Jdk8Module`? See https://github.com/whiskeysierra/problem#usage

File: README.md [hunk 1]
```diff
@@ -220,7 +223,8 @@ If you already have an exception class that you want to extend, you should imple
 public final class OutOfStockProblem extends BusinessException implements Exceptional
 ```
 
-The Jackson support module will recognize this inteface and deal with the inherited properties from `Throwable` accordingly. Note: This interface only exists, because `Throwable` is a concrete class, rather than an interface.
+The Jackson support module will recognize this inteface and deal with the inherited properties from `Throwable`
```
Comment (lukasniemeier) [inline README.md]: Typo "inteface"

Comment (whiskeysierra) [inline README.md]: > Typo "inteface"

Fixed

**Comment by codechimp** [comment]
@whiskeysierra looks good to me :+1:

**Comment by lukasniemeier-zalando** [comment]
:+1: top!

**Pull request merged**